src/scripts/app.ts

```diff
@@ -584,6 +585,90 @@ export class ComfyApp {
+   */
+  #addContextMenuTranslator() {
+    // 右键上下文菜单
+    const f = LGraphCanvas.prototype.getCanvasMenuOptions;
+    LGraphCanvas.prototype.getCanvasMenuOptions = function () {
+      var res = f.apply(this, arguments);
+      for (let item of res) {
+        if (item == null || !item.hasOwnProperty("content")) continue;
+        item.content = st(`graphCanvasMenu.${item.content}`, item.content);
+      }
+      return res;
+    };
+
+    function translateMenus(values, options)
+    {
+      if (!values) return;
+      // Convert {w.name} to input
+      // Convert {w.name} to widget
+      var reInput = /Convert (.*) to input/;
+      var reWidget = /Convert (.*) to widget/;
+      var cvt = st("graphCanvasMenu.Convert ", "Convert ");
+      var tinp = st("graphCanvasMenu. to input", " to input");
+      var twgt = st("graphCanvasMenu. to widget", " to widget");
+      for (var i = 0; i < values.length; i++) {
+        var value = values[i];
+        // 这里对应 从 slot 拖拽弹出的菜单, 但翻译后 菜单不能正常工作
+        // if (typeof value === 'string') {
+        //   values[i] = st(`graphCanvasMenu.${value}`, value);
+        //   continue;
+        // }
+        translateMenus(value?.submenu?.options, options);
+        if (value == null || !value.hasOwnProperty("content")) continue;
+        if (te(`graphCanvasMenu.${value.content}`)){
+          value.content = st(`graphCanvasMenu.${value.content}`, value.content);
+          continue;
+        }
+
+        // for capture translation text of input and widget
+        var extra_info = options.extra || options.parentMenu?.options?.extra;
+        // widgets and inputs
+        var matchInput = value.content?.match(reInput);
+        if (matchInput) {
+          var match = matchInput[1];
+          extra_info?.inputs?.find((i) => {
+            if (i.name != match) return false;
+            match = i.label ? i.label : i.name;
+          });
+          extra_info?.widgets?.find((i) => {
+            if (i.name != match) return false;
+            match = i.label ? i.label : i.name;
+          });
+          value.content = cvt + match + tinp;
+          continue;
+        }
+        var matchWidget = value.content?.match(reWidget);
+        if (matchWidget) {
+          var match = matchWidget[1];
+          extra_info?.inputs?.find((i) => {
+            if (i.name != match) return false;
+            match = i.label ? i.label : i.name;
+          });
+          extra_info?.widgets?.find((i) => {
+            if (i.name != match) return false;
+            match = i.label ? i.label : i.name;
+          });
+          value.content = cvt + match + twgt;
+          continue;
+        }
+      }
+    }
+
+    const OriginalContextMenu = LiteGraph.ContextMenu;
+    LiteGraph.ContextMenu = function (values, options) {
+      if (options.hasOwnProperty("title")) {
+        options.title = st(`nodeDefs.${normalizeI18nKey(options.title)}.display_name`, options.title);
```

Comment:
Before:
![image](https://github.com/user-attachments/assets/c8102a2a-5eea-4bff-8045-e0d93d68b8c8)
After:
![image](https://github.com/user-attachments/assets/bd9644d3-0571-4857-983c-3522d15f8184)

I think this is an acceptable change. We can update the playwright test expectations.